### ISSUE TITLE
Can't use actions, even if copied rules from documentation

### ISSUE THREAD

**Issue by PuppyNami** [tfeldmann/organize#352]
**Describe the bug**
When using actions like move, trash and delete, not tested the others, I get a validation error from my config that looks like this:
```
Error: Config problem
1 validation error in "C:\Users\UserName\AppData\Local\organize\organize\config.yaml"

rules.0.actions.0
  Input should be an instance of Action [input_value="Delete()", input_type=Delete]
```

**Environment (please complete the following information):**

OS: Microsoft Windows 11 Pro Version 10.0.22631 Build 22631
Python version: 3.12.2
Output of organize --version: organize, version 3.1.0

**Your config file**

```yaml
all_my_messy__download_folders: &download_folders
  - D:/Windows/Downloads/Chrome
  - D:/Windows/Downloads/Chrome/Other

rules:
  - name: "Delete Duplicate"
    locations: *download_folders
    filters:
      - duplicate:
          detect_original_by: "created"
    actions:
      - delete
```

**Comment by tfeldmann** [comment]
Thank you for reporting! This is a strange one, your config file is working fine here.
Are you sure you are editing the correct config file? Could you attach this file here (or send an email) so I can look for encoding problems?

**Comment by PuppyNami** [comment]
![image](https://github.com/tfeldmann/organize/assets/58266091/0814bd72-d39a-42a1-8504-67f0cc763231)

**Comment by tfeldmann** [comment]
Thank you. What‘s the output of`organize debug`?

**Comment by PuppyNami** [comment]
`organize debug` seems to be the same output as `organize check`

**Comment by PuppyNami** [comment]
I tested a little bit more and it seems to be only happening with the latest 3.12 python version 
![image](https://github.com/tfeldmann/organize/assets/58266091/e6e57ba2-dcdc-4537-be7e-933e3310a7e8)

**Comment by tfeldmann** [comment]
Thanks for debugging! I just tried Python 3.12.1 and failed to reproduce this. What a strange bug.

**Comment by tfeldmann** [comment]
Got a reproducer: 
```python
"""
docker run --rm -v.:/app --entrypoint=python organize3.12 /app/reproducer.py
docker run --rm -v.:/app --entrypoint=python organize3.12.1 /app/reproducer.py

"""
from dataclasses import dataclass

from typing_extensions import Protocol, runtime_checkable


@runtime_checkable  # removing this decorator fixes the bug.
class HasConfig(Protocol):
    pass

@runtime_checkable
class Action(HasConfig, Protocol):
    pass

@dataclass
class Delete:
    pass

delete = Delete()
print(isinstance(delete, Action))
```

This prints `False` in 3.12.2, `True` in 3.12.1.

**Comment by tfeldmann** [comment]
I raised an upstream issue here: https://github.com/python/typing_extensions/issues/335

**Comment by tfeldmann** [comment]
Thanks for reporting. I just released v3.1.2. Can you try again with this version?

**Comment by tfeldmann** [comment]
Thanks for the fast response, thats good to hear!
